packages/frontend/src/components/connect-wallet/index.tsx

```diff
@@ -10,36 +10,49 @@ import { ChainIcon } from "../chain-icon";
 import { NetworksPopover } from "./popovers/networks";
 import { ConnectPopover } from "./popovers/connect";
 import { AccountPopover } from "./popovers/account";
-import { useClickAway } from "react-use";
 import { Avatar } from "./avatar";
 
 export const ConnectWallet = () => {
     const { t } = useTranslation();
     const { chain } = useNetwork();
     const { address, connector: activeConnector } = useAccount();
 
-    const networksPopoverAnchorRef = useRef<HTMLDivElement>(null);
+    const [networksPopoverAnchor, setNetworksPopoverAnchor] =
+        useState<HTMLDivElement | null>(null);
+    const [connectWallet, setConnectWallet] =
+        useState<HTMLButtonElement | null>(null);
+
     const networksPopoverRef = useRef<HTMLDivElement>(null);
-    const connectWalletRef = useRef<HTMLButtonElement>(null);
     const connectPopoverRef = useRef<HTMLDivElement>(null);
     const accountPopoverRef = useRef<HTMLDivElement>(null);
 
     const [networksPopoverOpen, setNetworksPopoverOpen] = useState(false);
     const [connectPopoverOpen, setConnectPopoverOpen] = useState(false);
     const [accountPopoverOpen, setAccountPopoverOpen] = useState(false);
 
-    useClickAway(networksPopoverRef, () => {
-        setNetworksPopoverOpen(false);
-    });
-
-    useClickAway(connectPopoverRef, () => {
-        setConnectPopoverOpen(false);
-    });
-
-    useClickAway(accountPopoverRef, () => {
-        setAccountPopoverOpen(false);
-    });
-
+    useEffect(() => {
+        const handleClick = (event: MouseEvent) => {
```

Comment:
Should we rename this `handleMouseDown` since it's reacting to `mousedown` events?

Reply:
Addressed in 7fbbd22342ede86656644d348431278e24706b30